### ISSUE TITLE
Request to add a top private domain mode as seen in google's guava.

### ISSUE THREAD

**Issue by t0xodile** [tomnomnom/unfurl#25]
https://guava.dev/releases/32.1.1-jre/api/docs/com/google/common/net/InternetDomainName.html#topPrivateDomain()

Would be very helpful. "Returns the portion of this domain name that is one level beneath the [public suffix](https://guava.dev/releases/32.1.1-jre/api/docs/com/google/common/net/InternetDomainName.html#isPublicSuffix()). For example, for x.adwords.google.co.uk it returns google.co.uk, since co.uk is a public suffix. Similarly, for myblog.blogspot.com it returns the same domain, myblog.blogspot.com, since blogspot.com is a public suffix."

[Public Suffixes](https://publicsuffix.org/list/public_suffix_list.dat). Apex works in some cases but not all. 

Basically the mode should check the above list, and then pull out the domain right before that. A good example is 
something.somethingelse.azurewebsites.net.

Apex mode would output azurewebsites.net.
Top private domain mode, would output somethingelse.azurewebsites.net.

Hope that makes sense.
\\Tom
